### ISSUE TITLE
Run time error

### ISSUE THREAD

**Issue by YangXS** [zhuhan1236/mmd-caffe#2]
I have successfully complied the project. But run time error appears when I start training with the command：
build/tools/caffe.bin train -solver models/icml/amazon_to_webcam_common_setting/solver.prototxt -weights models/bvlc_reference_caffenet.caffemodel -gpu 0

I0915 14:21:56.463819 27231 mmd_layer.cu:69] before mmd diff 5.76657
I0915 14:21:56.468710 27231 mmd_layer.cu:107] bandwidth 3.65977e-05
I0915 14:21:56.476428 27231 mmd_layer.cu:464] after mmd diff sum 7.71586
I0915 14:21:56.476439 27231 mmd_layer.cu:465] ------
I0915 14:21:56.486502 27231 solver.cpp:189] Iteration 8, loss = 2.62049
I0915 14:21:56.486516 27231 solver.cpp:465] Iteration 8, lr = 0.001
F0915 14:21:56.619989 27231 math_functions.cu:125] Check failed: status == CUBLAS_STATUS_SUCCESS (11 vs. 0)  CUBLAS_STATUS_MAPPING_ERROR
**\* Check failure stack trace: ***
    @     0x7f95c82e25cd  google::LogMessage::Fail()
    @     0x7f95c82e4433  google::LogMessage::SendToLog()
    @     0x7f95c82e215b  google::LogMessage::Flush()
    @     0x7f95c82e4e1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f95c87579de  caffe::caffe_gpu_asum<>()
    @     0x7f95c8792329  caffe::SoftmaxWithLossLayer<>::Forward_gpu()
    @     0x7f95c8742712  caffe::Net<>::ForwardFromTo()
    @     0x7f95c8742907  caffe::Net<>::ForwardPrefilled()
    @     0x7f95c8731042  caffe::Solver<>::Step()
    @     0x7f95c8731b64  caffe::Solver<>::Solve()
    @           0x407e99  train()
    @           0x405be8  main
    @     0x7f95c7775830  __libc_start_main
    @           0x406109  _start
    @              (nil)  (unknown)
Aborted (core dumped)

**Comment by zhuhan1236** [comment]
It seems that there is something wrong with CUBLAS. Did you use `train.txt` file given in the project as training input?

**Comment by YangXS** [comment]
@zhuhan1236 ,
Thanks for your reply. 
Yes, I use the given train.txt file in the train_val.prototxt

**Comment by YangXS** [comment]
@zhuhan1236 
Now, the code can run by adding the scale parameters in the prototxt file.
However, using the default setting, the result for the amazon to webcam task is:
Iteration 100000, loss = 0.683849
I1102 07:26:31.289079 18809 solver.cpp:267] Iteration 100000, Testing net (#0)
I1102 07:26:34.337460 18809 solver.cpp:316]     Test net output #0: accuracy = 0.309434

How to improve the performance ?

**Comment by zhuhan1236** [comment]
Where did you add scale parameters? I couldn't find it in my prototxt file. And the loss value at iteration 100000 seems to be too large, which means amazon itself may have a high classification error. Maybe there's something wrong with scale parameters?

**Comment by YangXS** [comment]
@zhuhan1236 , Thanks for your reply. I use the following scale parameter.
transform_param {
    scale: 0.00390625
    crop_size: 227
    mirror: true
    mean_value: 104.0
    mean_value: 117.0
    mean_value: 123.0
  }

If I remove the scale parameter, the run time error appears again.

**Comment by zhuhan1236** [comment]
I'm sorry that I've check the source code carefully and I haven't find how param 'scale' related to the above error message. According to the log message, it's the 'caffe_gpu_asum' function in line 53 or line 56 of 'softmax_loss_layer.cu' issues an error. Would you please remove 'scale' param and print some debug info before that? Maybe 'nthread' and the shape of 'loss_data' or 'counts'. Thank you

**Comment by hemanthdv** [comment]
I am facing the same problem. Has this error been resolved? What is the solution?

I1106 17:41:24.313314 23781 mmd_layer.cu:69] before mmd diff 6.11412
I1106 17:41:24.315937 23781 mmd_layer.cu:107] bandwidth 4.99099e-05
I1106 17:41:24.325299 23781 mmd_layer.cu:464] after mmd diff sum 8.20216
I1106 17:41:24.325317 23781 mmd_layer.cu:465] ------
I1106 17:41:24.333673 23781 solver.cpp:189] Iteration 7, loss = 3.46312
I1106 17:41:24.333694 23781 solver.cpp:465] Iteration 7, lr = 0.001
F1106 17:41:24.495678 23781 math_functions.cu:125] Check failed: status == CUBLAS_STATUS_SUCCESS (11 vs. 0)  CUBLAS_STATUS_MAPPING_ERROR
**\* Check failure stack trace: ***
    @     0x7f6dcd3c05cd  google::LogMessage::Fail()
    @     0x7f6dcd3c2433  google::LogMessage::SendToLog()
    @     0x7f6dcd3c015b  google::LogMessage::Flush()
    @     0x7f6dcd3c2e1e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f6dcd89a17e  caffe::caffe_gpu_asum<>()
    @     0x7f6dcd87c299  caffe::SoftmaxWithLossLayer<>::Forward_gpu()
    @     0x7f6dcd7f0be2  caffe::Net<>::ForwardFromTo()
    @     0x7f6dcd7f0dd7  caffe::Net<>::ForwardPrefilled()
    @     0x7f6dcd73a652  caffe::Solver<>::Step()
    @     0x7f6dcd73b174  caffe::Solver<>::Solve()
    @           0x407e99  train()
    @           0x405be8  main
    @     0x7f6dcc853830  __libc_start_main
    @           0x406109  _start
    @              (nil)  (unknown)
Aborted (core dumped)

**Comment by zhuhan1236** [comment]
Haven't resolved yet because I can not reproduce the error. I searched it in google and found that add 'scale' param like above may eliminate the error but cause large accuracy drop. Would you please give me more runtime log message?  'nthread' and the shape of 'loss_data' or 'counts' , e.g.

**Comment by hemanthdv** [comment]
Can you please tell how to extract that information?

**Comment by zhuhan1236** [comment]
I've just pushed a debug branch. Please clone and make it.

**Comment by hemanthdv** [comment]
Thanks. I did that. My `make runtest` fails after caffe build. But the mmd program seems to be running ok now.

**Comment by zhuhan1236** [comment]
That's fine. I'm sorry that the implementation is not elegant enough. I'm
going to publish a new version which is more 'caffe-like' in the future.

**Comment by zhuhan1236** [comment]
Please move to [transfer-caffe](https://github.com/zhuhan1236/transfer-caffe)
